scalalib/src/mill/javalib/spotless/settings.scala

```diff
@@ -0,0 +1,224 @@
+        steps.add(ImportOrderStep.forJava().createFrom(groups.toArray*))
+      case Some(_) =>
+        steps.add(ImportOrderStep.forJava().createFrom(Array.empty[String]*))
+      case None =>
+        importOrderFile.foreach { file =>
+          steps.add(ImportOrderStep.forJava().createFrom(getConfFile(millSourcePath, file)))
+        }
+    }
+
+    if (removeUnusedImports) {
+      steps.add(RemoveUnusedImportsStep.create(
+        RemoveUnusedImportsStep.defaultFormatter(),
+        provisioner
+      ))
+    }
+
+    if (cleanthat) {
+      steps.add(CleanthatJavaStep.create(provisioner))
+    }
+
+    formatter.foreach {
+      case gf: GoogleJavaFormat =>
+        val step = GoogleJavaFormatStep.create(
+          gf.groupArtifact,
+          gf.version,
+          if (gf.aosp) "AOSP" else "GOOGLE",
+          provisioner,
+          gf.reflowLongStrings,
+          gf.reorderImports,
+          gf.formatJavadoc
+        )
+        steps.add(step)
+
+      case pjf: PalantirJavaFormat =>
+        val step = PalantirJavaFormatStep.create(pjf.version, pjf.style, provisioner)
+        steps.add(step)
+    }
+
+    if (formatAnnotations) {
+      steps.add(FormatAnnotationsStep.create())
+    }
+
+    super.addLicenseHeaderStep(millSourcePath, steps)
+
+    steps
+  }
+}
+
+case class ScalaConfig(
+    target: String = ".scala",
+    licenseHeader: Option[String] = None,
+    licenseHeaderFile: Option[String] = None,
+    scalafmtLibVersion: String = "3.8.1",
+    scalafmtConfigFile: Option[String] = None,
+    scalafmtScalaMajorVersion: String = "2.13"
+) extends JVMLangConfig {
+  require(
+    !(licenseHeader.isDefined && licenseHeaderFile.isDefined),
+    "Please specify only licenseHeader or licenseHeaderFile but not both"
+  )
+
+  def getSteps(millSourcePath: os.Path, provisioner: Provisioner): List[FormatterStep] = {
+    val steps = new ArrayList[FormatterStep]()
+
+    scalafmtConfigFile.map(file => {
+      val configFile = getConfFile(millSourcePath, file)
+      val scalafmtStep = ScalaFmtStep.create(
+        scalafmtLibVersion,
+        scalafmtScalaMajorVersion,
+        provisioner,
+        configFile
+      )
+      steps.add(scalafmtStep)
+    })
+
+    super.addLicenseHeaderStep(millSourcePath, steps)
+
+    steps
+  }
+}
+
+case class KtfmtOptions(
+    maxWidth: Option[Int] = Some(80),
+    blockIndent: Option[Int] = Some(4),
+    continuationIndent: Option[Int] = Some(4),
+    removeUnusedImports: Option[Boolean] = Some(false),
+    manageTrailingCommas: Option[Boolean] = Some(false)
+)
+case class KotlinConfig(
+    target: String = ".kt",
+    licenseHeader: Option[String] = None,
+    licenseHeaderFile: Option[String] = None,
+    override val licenseHeaderDelimiter: String = "(package |@file|import )",
+    ktfmtFlag: Boolean = false,
+    ktfmtVersion: String = "0.53",
+    ktfmtOptions: Option[KtfmtOptions] = None,
+    ktlintFlag: Boolean = false,
+    ktlintVersion: String = "1.5.0"
```

Comment:
Don't hardcode versions. Theese are already available via the `Versions` object.